pkg/core/blockchain.go

```diff
@@ -679,9 +716,55 @@ func (bc *Blockchain) GetStateModule() blockchainer.StateRoot {
 	return bc.stateRoot
 }
 
+// saveBlock stores specified block skipping executable scripts and updates
+// Blockchain state.
+func (bc *Blockchain) saveBlock(block *block.Block) error {
+	cache := dao.NewCached(bc.dao)
+	writeBuf := io.NewBufBinWriter()
+	if err := cache.StoreAsBlock(block, writeBuf); err != nil {
+		return err
+	}
+	writeBuf.Reset()
+
+	if err := cache.StoreAsCurrentBlock(block, writeBuf); err != nil {
+		return err
+	}
+	writeBuf.Reset()
+
+	for _, tx := range block.Transactions {
+		if err := cache.StoreAsTransaction(tx, block.Index, writeBuf); err != nil {
+			return err
+		}
+		writeBuf.Reset()
+
+		err := bc.storeConflictTxs(tx, block, writeBuf, cache)
+		if err != nil {
+			return err
+		}
+	}
+	if bc.config.SaveStorageBatch {
+		bc.lastBatch = cache.DAO.GetBatch()
```

Comment:
It doesn't make sense, batches are for storage changes and storage isn't changed here.